packages/nimble-components/src/table-column/date-text/tests/table-column-date-text.spec.ts

```diff
@@ -26,187 +26,558 @@ async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
+                0,
+                0,
+                new MouseEvent('mouseover')
+            );
+            await waitForUpdatesAsync();
+            expect(pageObject.getCellTitle(0, 0)).toEqual('');
+        });
 
-    it('when no fieldName provided, nothing is displayed', async () => {
-        await connect();
-        await waitForUpdatesAsync();
+        it('removes title on mouseout of cell', async () => {
+            element.style.width = '200px';
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            tablePageObject.dispatchEventToCell(
+                0,
+                0,
+                new MouseEvent('mouseover')
+            );
+            await waitForUpdatesAsync();
+            tablePageObject.dispatchEventToCell(
+                0,
+                0,
+                new MouseEvent('mouseout')
+            );
+            await waitForUpdatesAsync();
+            expect(pageObject.getCellTitle(0, 0)).toEqual('');
+        });
 
-        const firstColumn = element.columns[0] as TableColumnDateText;
-        firstColumn.fieldName = undefined;
-        await element.setData([
-            { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
-        ]);
-        await waitForUpdatesAsync();
+        it('sets group header text to rendered date value', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedGroupHeaderContent(0)).toBe(
+                'Dec 10, 2012, 10:35:05 PM'
+            );
+        });
 
-        expect(pageObject.getRenderedCellContent(0, 0)).toEqual('');
-    });
+        it('updates displayed date when format changes', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            column.format = 'custom';
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe('12/10/2012');
+        });
 
-    it('sets title when cell text is ellipsized', async () => {
-        element.style.width = '200px';
-        await element.setData([
-            { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
-        ]);
-        await connect();
-        await waitForUpdatesAsync();
-        tablePageObject.dispatchEventToCell(0, 0, new MouseEvent('mouseover'));
-        await waitForUpdatesAsync();
-        expect(pageObject.getCellTitle(0, 0)).toEqual(
-            'Dec 10, 2012, 10:35:05 PM'
-        );
-    });
+        it('honors customDateStyle property', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
```

Comment:
In the past I've seen date tests be flaky when run on machines in different time zones or locales. Is there anything we need to do to ensure these pass reliably?

I didn't think too much about it but as long as the way we construct dates and the way we format dates both use the same time zone (either local or UTC), then time zones shouldn't be an issue. Could you check that's the case?

And until we implement a way to set locale I think the DateTimeFormatter will also use the machine's locale. If that's the case then some of these tests may fail when the OS setting isn't en-us. Could you either try that or think through whether it's possible?

If any of them don't pass in other locations then we can talk about options. We may be willing to live with it until we implement the design token for setting locale. Or we may want to see if we can add a workaround for now.

Reply:
There's only one test that sets a specific output time zone, and I've already updated the input time string to include that same time zone. The rest of the tests will use the same (local) time zone for the input and output dates, so there shouldn't be a problem when running in different time zones.

On the other hand, the tests expect specific, locale-dependent, formatted strings. This breaks if the locale is something other than 'en-us'. I think we will want to explicitly set the locale to 'en-us' in the tests once we support that. I don't think we need to do anything special until then, though.

Reply:
I have my locale set to `en-ca` (Canada) which uses "p.m." instead of "PM". When I cloned your branch and ran the tests locally, they passed! I haven't debugged how that's possible: playing with DateTimeFormat on mdn definitely shows it using my local machine locale. Anyway, I agree that it's ok to keep as-is until we add the locale feature.

Reply:
In Windows, I found that they all pass if I just change my "Country or region" to Canada (with "Regional format" set to Recommended), but if I explicitly set my "Regional format" to English (Canada), the tests failed. How are you setting your locale? Is it just for the OS, or for all programs?

Reply:
I have it set in macOS System Preferences. I'm not sure how consistent apps are at following that.

Further confusing matters

- the sample dates in the settings dialog use AM 
- the DateTimeFormatter with a locale of `en-ca` or `undefined` in Safari uses AM
- the DateTimeFormatter with a locale of `en-ca` or `undefined` in Firefox uses a.m.
 
I think we should stop looking at this before we get sucked in any deeper :-D

<img width="827" alt="image" src="https://github.com/ni/nimble/assets/10500124/26f7de82-53f5-4236-a89f-0295a2e70c8b">